specification/resources/floating_ips/models/floating_ip_actions.yml

```
@@ -1,31 +1,34 @@
 floating_ip_action_type:
-  type: string
-  enum:
-  - assign
-  - unassign
-  example: assign
-  description: The type of action to initiate for the floating IP.
-
-floating_ip_action_unassign:
   type: object
-  properties:
-    type:
-      $ref: '#/floating_ip_action_type'
-  additionalProperties: false
   required:
-  - type
-
-floating_ip_action_assign:
-  type: object
-  properties:
-    type:
-      $ref: '#/floating_ip_action_type'
+    - type
+  discriminator:
```

```suggestion
  properties:
    type:
      type: string
      description: The type of action to initiate for the floating IP.
  discriminator:
```

Since we're using allOf with `$ref: '#/floating_ip_action_type'` in the other objects, if we add this here we do not need to repeat 

```
        type:
          type: string
```

in both of them. It also retains the description for `type`.